lib/doro/comms.ex

```diff
@@ -3,7 +3,7 @@ defmodule Doro.Comms do
   Functions for sending output to players.
   """
   def send_to_player(player = %Doro.Entity{}, s) do
-    Phoenix.PubSub.broadcast(Doro.PubSub, "player-session:#{player.id}", {:send, s})
+    send_output_to_player(player, s)
```

Comment:
Would it be simpler to augment the public methods in this module to take an optional `data` parameter?  That way, the "textual" representation would always exist, and you could optionally add structured data. Something like:

```elixir
def send_to_player(player, s, data \\ %{}) do
  payload = %{text: s, data: data}  # or Map.put(data, :text, s)
  broadcast(Doro.PubSub, "player-session:#{player.id}", {:send, payload})
end
```

So a command "/edit sewing machine" would do the correct thing on all clients:
```elixir
send_to_player(p, "You concentrate on #{definite(entity)}, until you feel silly.", %{type: :edit, entity: marshal(entity)})
```

With this, you'd have to also modify the [slack session](https://github.com/dorodango/doro/blob/8b136d7722c9c5523ba43ec54d3a7cd27357c4a1/lib/doro/transports/slack/session.ex#L29-L30) and the [web session](https://github.com/dorodango/doro/blob/8b136d7722c9c5523ba43ec54d3a7cd27357c4a1/lib/doro/session.ex#L41-L42) to use the new payload.

Reply:
yep.  sounds good.  as we discussed in person, better to have a response for UI's that don't support the action.